From 1405f5d5a978b66855692618af468c9f2ded4e21 Mon Sep 17 00:00:00 2001
From: Nek Saikou <NekSaikou@proton.me>
Date: Fri, 24 May 2024 17:45:01 +0700
Subject: [PATCH 110/228] ecstresstest.c: Fix memory leak on error

Fixes #24476
CLA: trivial

Reviewed-by: Neil Horman <nhorman@openssl.org>
Reviewed-by: Tomas Mraz <tomas@openssl.org>
(Merged from https://github.com/openssl/openssl/pull/24488)

(cherry picked from commit 434e7f7cb4259f8c8c1463fd38fe723b3efca887)
---
 test/ecstresstest.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/ecstresstest.c b/test/ecstresstest.c
index 22d46c50da..b4dd463740 100644
--- a/test/ecstresstest.c
+++ b/test/ecstresstest.c
@@ -79,7 +79,7 @@ static int test_curve(void)
             || !TEST_ptr(point = EC_POINT_dup(EC_GROUP_get0_generator(group),
                                               group))
             || !TEST_ptr(result = walk_curve(group, point, num_repeats)))
-        return 0;
+        goto err;
 
     if (print_mode) {
         BN_print(bio_out, result);
-- 
2.39.5 (Apple Git-154)

